### PR TITLE
Add unit tests for code coverage

### PR DESCRIPTION
**Pull request by jahid1209**
- Added unit tests to improve code coverage from 54% to 72%.

### CODE REVIEW

**Review by zaman-akib — APPROVED** [review]

File: src/main/java/com/synopsys/integration/jenkins/scan/factory/ScanParametersFactory.java [hunk 1]
```diff
@@ -48,8 +48,8 @@ public static PluginParametersHandler createPipelineCommand(Run<?, ?> run, TaskL
                         new ScannerArgumentService(listener, envVars, workspace)), workspace, envVars, listener);
     }
 
-    public static Map<String, Object> preparePipelineParametersMap(SecurityScanStep scanStep, FilePath workspace, TaskListener listener) throws PluginExceptionHandler {
-        Map<String, Object> parametersMap = new HashMap<>(getGlobalConfigurationValues(workspace, listener));
+    public static Map<String, Object> preparePipelineParametersMap(SecurityScanStep scanStep, Map<String, Object> parametersMap,
+                                                                   FilePath workspace, TaskListener listener) throws PluginExceptionHandler {
```
[the remark this answers is not in the document]
Comment (jahid1209) [reply src/main/java/com/synopsys/integration/jenkins/scan/factory/ScanParametersFactory.java]: Nice catch!

File: src/test/java/com/synopsys/integration/jenkins/scan/factory/ScanParametersFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        Map<String, Object> globalConfigValues = new HashMap<>();
+        ScanParametersFactory scanParametersFactory = new ScanParametersFactory(envVarsMock, workspace);
+
+        securityScanStep.setProduct("BLACKDUCK");
+        securityScanStep.setBitbucket_token("FAKETOKEN");
+        globalConfigValues.put(ApplicationConstants.BLACKDUCK_URL_KEY, "https://fake-blackduck.url");
+        globalConfigValues.put(ApplicationConstants.BLACKDUCK_TOKEN_KEY, "fake-blackduck-token");
+        globalConfigValues.put(ApplicationConstants.SYNOPSYS_BRIDGE_INSTALL_DIRECTORY, "/fake/path");
+
+        Map<String, Object> result = scanParametersFactory.preparePipelineParametersMap(securityScanStep,
```
Comment (maksudur-rahman-maruf) [inline src/test/java/com/synopsys/integration/jenkins/scan/factory/ScanParametersFactoryTest.java:45]: Here, `preparePipelineParametersMap` static method is accessed by the instance reference. I think static method should be accessed by the class name.

File: src/test/java/com/synopsys/integration/jenkins/scan/factory/ScanParametersFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+        assertEquals(5, result.size());
+        assertEquals("BLACKDUCK", result.get(ApplicationConstants.PRODUCT_KEY));
+        assertEquals("fake-blackduck-token", result.get(ApplicationConstants.BLACKDUCK_TOKEN_KEY));
+        assertEquals("/fake/path", result.get(ApplicationConstants.SYNOPSYS_BRIDGE_INSTALL_DIRECTORY));
+        assertEquals("FAKETOKEN", result.get(ApplicationConstants.BITBUCKET_TOKEN_KEY));
+
+        securityScanStep.setProduct("invalid-product");
+
+        assertThrows(PluginExceptionHandler.class, () -> scanParametersFactory.preparePipelineParametersMap(securityScanStep,
```
Comment (maksudur-rahman-maruf) [inline src/test/java/com/synopsys/integration/jenkins/scan/factory/ScanParametersFactoryTest.java:56]: Here, preparePipelineParametersMap static method is accessed by the instance reference. I think static method should be accessed by the class name.

File: src/test/java/com/synopsys/integration/jenkins/scan/factory/ScanParametersFactoryTest.java [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        securityScanStep.setBlackduck_install_directory("/fake/path");
+        securityScanStep.setBlackduck_scan_full(true);
+        securityScanStep.setBlackduck_automation_prcomment(true);
+        securityScanStep.setBlackduck_download_url("https://fake.blackduck-download-url");
+        securityScanStep.setBlackduck_scan_failure_severities("MAJOR");
+
+        Map<String, Object> blackDuckParametersMap = ScanParametersFactory.prepareBlackDuckParametersMap(securityScanStep);
+
+        assertEquals(7, blackDuckParametersMap.size());
+        assertEquals("https://fake.blackduck-url", blackDuckParametersMap.get(ApplicationConstants.BLACKDUCK_URL_KEY));
```
Comment (maksudur-rahman-maruf) [inline src/test/java/com/synopsys/integration/jenkins/scan/factory/ScanParametersFactoryTest.java:73]: We are using the same string values in multiple places. I think we should define  these values in  variables and then access those from the variables. 
For example, we are repeating this `"https://fake.blackduck-url"`

File: src/test/java/com/synopsys/integration/jenkins/scan/factory/ScanParametersFactoryTest.java [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+        securityScanStep.setCoverity_policy_view("fake-policy");
+        securityScanStep.setCoverity_install_directory("/fake/path");
+        securityScanStep.setCoverity_automation_prcomment(true);
+        securityScanStep.setCoverity_version("1.0.0");
+        securityScanStep.setCoverity_local(true);
+
+        Map<String, Object> coverityParametersMap = ScanParametersFactory.prepareCoverityParametersMap(securityScanStep);
+
+        assertEquals(10, coverityParametersMap.size());
+        assertEquals("https://fake.coverity-url", coverityParametersMap.get(ApplicationConstants.COVERITY_URL_KEY));
```
Comment (maksudur-rahman-maruf) [inline src/test/java/com/synopsys/integration/jenkins/scan/factory/ScanParametersFactoryTest.java:102]: Same for coverity too.

File: src/test/java/com/synopsys/integration/jenkins/scan/factory/ScanParametersFactoryTest.java [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+        securityScanStep.setPolaris_application_name("fake-application-name");
+        securityScanStep.setPolaris_project_name("fake-project-name");
+        securityScanStep.setPolaris_assessment_types("SCA");
+        securityScanStep.setPolaris_triage("REQUIRED");
+        securityScanStep.setPolaris_branch_name("test");
+
+        Map<String, Object> polarisParametersMap = ScanParametersFactory.preparePolarisParametersMap(securityScanStep);
+
+        assertEquals(7, polarisParametersMap.size());
+        assertEquals("https://fake.polaris-server.url", polarisParametersMap.get(ApplicationConstants.POLARIS_SERVER_URL_KEY));
```
Comment (maksudur-rahman-maruf) [inline src/test/java/com/synopsys/integration/jenkins/scan/factory/ScanParametersFactoryTest.java:152]: Same for polaris.

**Pull request merged**